docs/specifications/tests/Syntax-TP/syntax06.md

```diff
@@ -1,54 +1,129 @@
-## SYNTAX06: No illegal characters in the SOA RNAME field
+# SYNTAX06: No illegal characters in the SOA RNAME field
 
-### Test case identifier
-**SYNTAX06** There must be no illegal characters in the SOA RNAME field
+## Test case identifier
+**SYNTAX06**
 
-### Objective
+## Objective
 
 The SOA RNAME field is a mailbox address. The SOA RNAME field is defined
-in section 3.3 of [RFC 1034](https://tools.ietf.org/rfc/rfc1034.txt) and
-section 2.2 of [RFC 1912](https://tools.ietf.org/rfc/rfc1912.txt). The RNAME
-field should follow the rules of an e-mail address also defined in section
-3.4.1 of [RFC 2822](https://tools.ietf.org/html/rfc2822#section-3.4).
+in section 3.3.13 of [RFC 1035] and section 2.2 of [RFC 1912]. The RNAME
+field should follow the rules of an e-mail address also defined in 
+[RFC 5322, section 3.4.1].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+2. Create a SOA query for the apex of the *Child Zone* with RD flag unset.
+
+3. For each name server in *Name Server IP* do:
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond with a DNS response, then:
+      1. Output *[NO_RESPONSE]*.
+      2. Go to next server.
+   3. If the DNS response does not include a SOA record in the 
+      answer section, then:
+      1. Output *[NO_RESPONSE_SOA_QUERY]*.
+      2. Go to next server.
+   4. Extract the RNAME from the SOA record.
```

Comment:
It is illegal to return more than one SOA record for a zone. My experience is that it is very unusual that it happens, so that is why I have chosen to ignore that case. We query for SOA in many specifications. Should check for multiple SOA records in all cases?

Reply:
I agree that validation of the number of SOA records should only be validated by a single test case. Maybe we could have a comment to use the first SOA record in case more than one are present?

Reply:
A new test case has to be created (#666). Here we will just take the first. This is valid for many test cases.